src/format/scan.rs

```diff
@@ -219,9 +219,48 @@ pub(super) fn space1(s: &str) -> &str {
     }
 }
 
-/// Consumes any number (including zero) of colon or spaces.
+/// Allow a colon with possible one-character whitespace padding.
+/// Consumes zero or one of these leading patterns:
+/// `":"`, `" "`, `" :"`, `": "`, or `" : "`.
 pub(super) fn colon_or_space(s: &str) -> ParseResult<&str> {
```

Comment:
This functions return type is `ParseResult` but as far as I can see no code paths can return `Err`. We should either implement some `Err` code paths or have the function return just `&str` instead

Reply:
The `colon_or_space` is passed to `src/format/scan.rs:timezone_offset` functions.

https://github.com/chronotope/chrono/blob/v0.4.23/src/format/scan.rs#L219
Those functions require passing an object with trait
`FnMut(&str) -> ParseResult<&str>`.
It would touch too much code for this PR.

After a brief code review, I think `colon_or_space` should remain as is. Not quite as perfect as it could be but good enough IMO